libs/traits/src/lib.rs

```diff
@@ -125,6 +128,47 @@ pub trait PoolInspect<AccountId, CurrencyId> {
 	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 }
 
+/// Variants for valid Pool updates to send out as events
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub enum UpdateState {
+	NoExecution,
```

Comment:
I think the purpose of this was to cancel the outstanding scheduled update. So we should then add other functionality for achieving this.

Reply:
Yes, canceling should definitely be a dedicated extrinsics instead of implicit submitting all `NoChanges`.

Reply:
cc @gruberb

Reply:
I didn't change the logic of `pool-system` here: https://github.com/centrifuge/centrifuge-chain/blob/main/pallets/pool-system/src/lib.rs#L842-L846

This is why I added `NoExecution`, because currently we check inside `pool-system` if we just remove the scheduled updated and return `Ok(Some(T::WeightInfo::update_no_execution(0)).into());`.

Reply:
So you want me to adjust the `pool-system` logic and error out instead of returning `update_no_execution`?

Reply:
No, I would say lets do this in a follow up pr. Was not aware about the removing logic. This needs more changes then.